pkg/operator/operator.go

```diff
@@ -259,6 +264,76 @@ func (o *Operator) determineReplicasForDefaultIngressController() (int32, error)
 	return ingress.DetermineReplicas(ingressConfig, infraConfig), nil
 }
 
+// handleSingleNode4Dot11Upgrade sets the defaultPlacement status in the
+// ingress config CR of none-platform single node clusters to "ControlPlane" if
+// it's not already set. The situations in which this value is not set are in
+// clusters upgraded from <4.11 to >=4.11. When that field is not set, it is
+// assumed by this operator's controllers to have the value "Workers", which is
+// fine for most clusters, but not a good default for none-platform single-node
+// clusters as it causes issues when adding workers to them (and is why
+// starting with 4.11 freshly installed none-platform have it set to
+// "ControlPlane"). The purpose of this function is to then "fix" those older
+// clusters so they behave more like freshly installed >=4.11 clusters. Before
+// doing that we also verify that the cluster truly has only one node, as we
+// wouldn't want to affect clusters that have workers. That's because it's
+// likely that such clusters already have a custom ingress configuration that
+// we wouldn't want to potentially mess up, even though adding workers was
+// previously unsupported.
+// If defaultPlacement is unset and the cluster doesn't match the criteria
+// mentioned above, it is anyway given the explicit "Workers" value to ensure
+// this function doesn't affect it in the future no matter what changes (e.g.
+// number of nodes), and to make it more consistent with freshly installed
+// >=4.11 clusters.
+func (o *Operator) handleSingleNode4Dot11Upgrade() error {
+	ingressConfigName := types.NamespacedName{Name: "cluster"}
+
+	infraConfig := &configv1.Infrastructure{}
+	if err := o.client.Get(context.TODO(), ingressConfigName, infraConfig); err != nil {
```

Comment:
Using `ingressConfigName` for the infra config object's name works because the objects have the same name, but it looks strange.  Would you mind adding a function to `names.go` like `IngressClusterConfigName` and using that here?

Reply:
Ah absolutely, didn't even notice I did that